src/main/scala/lc/core/captcha.scala

```diff
@@ -1,38 +1,42 @@
 package lc.core
 
-import java.sql.{Blob, ResultSet}
+import java.sql.ResultSet
 import java.util.UUID
 import java.io.ByteArrayInputStream
 import lc.database.Statements
 import lc.core.CaptchaProviders
-import lc.captchas.interfaces.ChallengeProvider
+import java.nio.file.{Files, Paths}
+
 
 class Captcha {
 
   def getCaptcha(id: Id): Array[Byte] = {
-    var image: Array[Byte] = null
-    var blob: Blob = null
     try {
+      val imagePath = "data/404.jpg"
       val imagePstmt = Statements.tlStmts.get.imagePstmt
       imagePstmt.setString(1, id.id)
       val rs: ResultSet = imagePstmt.executeQuery()
-      if (rs.next()) {
-        blob = rs.getBlob("image")
-        if (blob != null) {
-          image = blob.getBytes(1, blob.length().toInt)
+      val image = if (rs.next()) {
+        val blob = rs.getBlob("image")
+        val imageBytes = if (blob != null) {
+          blob.getBytes(1, blob.length().toInt)
+        } else {
+          Files.readAllBytes(Paths.get(imagePath))
```

Comment:
Not sure this is a good idea. The calling application will not know that there was an error, but end user will see `404.jpg`.

I think it is better to return an internal error here, with status code 500.

Reply:
We were returning an empty image in those cases which appear broken on the client-side. Also, I remember, vaguely, that we decided to do something like this i.e, return an error image. 
But surely, throwing an error is better.